tfrecord/reader.py

```diff
@@ -69,17 +89,40 @@ def read_records(start_offset=None, end_offset=None):
     file.close()
 
 
-def tfrecord_loader(data_path, index_path, description, shard=None):
-    """Create an iterator from a tfrecord dataset. 
-
-    Args:
-        data_path: Path of the input data.
-        index_path: Path of index file. This can be set to None if not available.
-        description: A dictionary of key and values where keys are the name of the features and values correspond to
-                     data type. The data type can be "byte", "float" or "int".
-        shard: A tuple (index, count) representing the shard information. (default : None)
-    Returns:
-        An iterator that generates individual data records.
+def tfrecord_loader(data_path: str,
+                    description: typing.Dict[str, str],
```

Comment:
This breaks compatibility. Please revert the order of parameters. Also it's recommended to always provide index_path so it's good to explicitly set it to None when it's not needed.